clusterloader2/pkg/measurement/util/gatherers/container_resource_gatherer.go

```diff
@@ -103,12 +103,26 @@ func NewResourceUsageGatherer(c clientset.Interface, host string, port int, prov
 				return nil, fmt.Errorf("listing pods error: %v", err)
 			}
 		}
+
+		var nodeList *corev1.NodeList
+		var masterNodesMap map[string]bool = make(map[string]bool)
```

Comment:
Let's move after the list (if the list call fails you have unnecessary memory allocation).

Also, change to simply:
masterNodesMap := mak(map[string]bool)
or even better reuse string set:
https://github.com/kubernetes/perf-tests/blob/master/clusterloader2/vendor/k8s.io/apimachinery/pkg/util/sets/string.go

Reply:
Lessons are learned